main.cpp

```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+#include <vector>
+#include "Functions.h"
+#include "Animal.h"
+#include "Functions.cpp"
+#include "Animal.cpp"
+
+using namespace std;
+
+
+int main()
+{
+    vector<Animal> animals;
+    do
+    {
+        cout << "Ferma contine " << animals.size() << " animale." << endl;
+        switch(showMenu())
+        {
+            case 1:
+                addNewAnimal(animals);
+                break;
+            case 2: // Afiseaza intreaga lista a animalelor                
+                showEntireList(animals);                   
+                break;
+            case 3: // Modifica detaliile unui animal                                         
+                changeAnimalDetails(animals);
+                break;          
+            case 4: //  sterge un element din lista                
+                deleteAnimal(animals);                    
```

Comment:
nu stiu cat rost isi mai au comment-urile ca practic zic fix ce zice si functia de o apelezi. E un caz clasic de a scrie cod bun care e self-documenting. Codul in sine prin felul in care a fost scris si prin denumirile date functiilor este suficient de explicit incat nu mai ai nevoie de comentarii. e ceva de bine asta